akka-distributed-data/src/main/scala/akka/cluster/ddata/ORMap.scala

```diff
@@ -100,8 +179,18 @@ final class ORMap[A, B <: ReplicatedData] private[akka] (
         "`ORMap.put` must not be used to replace an existing `ORSet` " +
           "value, because important history can be lost when replacing the `ORSet` and " +
           "undesired effects of merging will occur. Use `ORMultiMap` or `ORMap.updated` instead.")
-    else
-      new ORMap(keys.add(node, key), values.updated(key, value))
+    else {
+      val putDeltaOp = PutDeltaOp(keys.resetDelta.add(node, key).delta.get, Map(key → value), node)
+      val newDelta = delta match {
+        case Some(d) ⇒
+          d.merge(putDeltaOp)
+        case None ⇒
+          putDeltaOp
+      }
+      // put forcibly damages history, so we remove added values from the tombstone
+      // and propagate full value that will overwrite previous values
+      new ORMap(keys.add(node, key), values.updated(key, value), tombstoneVals - key, Some(newDelta))
```

Comment:
Put forcibly damages history, let's not keep it in the tombstone, hence `tombstoneVals - key`